tests/cluecode/test_copyrights_fosso.py

```diff
@@ -134,9 +136,10 @@ def build_copyright_test_methods_with_fossology_data():
         tfn = test_file.replace(test_data_dir, '').strip('\/\\')
         test_name = 'test_fossology_copyright_%(tfn)s' % locals()
         test_name = python_safe_name(test_name)
-        if isinstance(test_name, unicode):
+        if isinstance(test_name, compat.unicode):
             test_name = test_name.encode('utf-8')
-
+        
+        test_name = test_name.decode() if py3 else test_name
```

Comment:
Rather than decoding... only encode above if on Python 2 e.g. something more or less like this would be better IMHO:
```
if py2 and isinstance(test_name, compat.unicode):
    test_name = test_name.encode('utf-8')
if py3 and not isinstance(test_name, compat.unicode):
    test_name = test_name.decode('utf-8')
```